doc/Developing.md

```diff
@@ -145,22 +163,45 @@ installing
 
 ## Build core, extensions and examples packages
 
-You can download dependencies and build it using:
+You can download dependencies and build TypeScript packages using:
 
-    cd $THEIA
-    yarn
+```sh
+cd $THEIA
+yarn
+```
 
-This command downloads dev dependencies, links and builds all packages.
-To learn more and understand precisely what's going on, please look at scripts in [package.json](../package.json).
+This command downloads dev dependencies, links and builds all TypeScript packages.
 
-### Skip lint
+To build the example applications:
 
-Linter can be skipped by setting the `SKIP_LINT` environment variable to `true`.
+```sh
+cd $THEIA
+yarn browser build
+yarn electron build
 
-example:
+# build both example applications at once:
+yarn build:examples
 ```
+
+To learn more and understand precisely what's going on, please look at scripts in [package.json](../package.json).
+
+## Build Everything
+
+```sh
+cd $THEIA
+yarn build
+```
```

Comment:
The only thing that gives me (slight) pause here is the change in behavior of the plain `yarn` command. If we want to have more options in how we build, there are basically two options: reduce `yarn`'s functionality and create a new command that has its old functionality (this PR), or leave `yarn` as was and create a new script with the reduced functionality of installing and transpiling. I don't think it matters much either way. Either we tell new people 'just `yarn` and then start' or we tell new people 'just `yarn build` and then start.' I might incline toward leaving `yarn` alone so that casual users' workflows don't change, but there's a certain iconicity to the shorter command (`yarn`) do less work than the longer command (`yarn build`).

Reply:
My biggest pain is that `yarn` alone is a shortcut to `yarn install` meaning it is meant to deal with dependencies. At best it is common practice to prepare the packages by transpiling sources, but linting and building two applications is overkill. This is especially annoying to work with when we need to work with dependencies when you have to keep an eye at the `yarn` output to see when to `ctrl+c` it.

With this PR, doing `yarn` will at least put the project in a state that can be published/consumed (we don't need to build the example apps to publish).

Reply:
More importantly: `yarn (install)` is the first command you HAVE to run in order to work with any npm package. Making that command do everything means there is no way to do less when preferable to do so.

Reply:
Sounds fair to me. 👍